lib/fielded_search.rb

```diff
@@ -0,0 +1,63 @@
+class FieldedSearch
+  attr_reader :field_pairs
+
+  def initialize(field_pairs)
+    @field_pairs = field_pairs
+  end
+
+  def present?
+    return false unless field_pairs
+    to_s.present?
+  end
+
+  def value_for(key)
+    if field_pairs && field_pairs.key?(key)
+      field_pairs[key]
+    end
+  end
+
+  def to_s
```

Comment:
`present?` is a method specific to Ruby on Rails, not Ruby.
```
[peterkarman@peterkarman-gsa:~/src/C2]$ irb
2.2.3 :001 > f = nil
 => nil 
2.2.3 :002 > f.present?
NoMethodError: undefined method `present?' for nil:NilClass
	from (irb):2
	from /Users/peterkarman/.rvm/rubies/ruby-2.2.3/bin/irb:15:in `<main>'
```
I chose that name intentionally however, because of how it is used in RoR, to indicate whether the object is not `blank?`.

The same is true of `to_h` and `to_s` -- those are special method names that Ruby programmers know, and they act consistently with this particular object type, which uses a delegation pattern. If you want a `FieldedSearch` object to stringify correctly and automatically, overriding `to_s` is the way to do it.